oggm/utils.py

```diff
@@ -1430,6 +1431,18 @@ def _divide_task(gdir, div_id=None, **kwargs):
         return _divide_task
 
 
+def self_iterating_entity_task(task_func):
+    """
```

Comment:
I think I'd rather give it some other name. "entity task" is a well defined concept in the (admittedly parse) documentation. In the API doc I called them ``global task``: http://oggm.readthedocs.io/en/latest/api.html#global-tasks

I'm open to other suggestions, though.